staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -177,6 +191,13 @@ func diffMaps(original, modified map[string]interface{}, t reflect.Type, diffOpt
 	}
 
 	updatePatchIfMissing(original, modified, patch, diffOptions)
+	// Insert the replaceKeysList when either of the following is true:
```

Comment:
Make comment match:
iff there are values present in the retainKeysList and either
- the patch is not empty
-